kubectl-plugin/test/e2e/kubectl_ray_log_test.go

```diff
@@ -0,0 +1,219 @@
+package e2e
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var requiredFileSet = map[string]string{
+	"stdout.log": "Ray runtime started",
+	"raylet.out": "Ray Event initialized for RAYLET",
+}
+
+var _ = Describe("Calling ray plugin `log` command on Ray Cluster", Ordered, func() {
+	It("succeed in retrieving all ray cluster logs", func() {
+		expectedDirPath := "./raycluster-sample"
```

Comment:
Should we first change current directory into a temp folder?

Reply:
A new directory is created for the raycluster and all the necessary logs should stay in there so I don't think there's a need to create a temp folder. Would creating a temp folder be safer/easier for cleanup? WDYT?

Reply:
I think it would be eaiser for cleanup. You only delete one folder at the end of the tests instead of using an `AfterEach` block.

Reply:
We would still need to do a cleanup after each 'It' block right? Since we want to make sure previous downloaded log files don't effect proceeding calls.

Reply:
Hmm. Actually I originally thought that some `It` didn't create folders, but you still executed `os.RemoveAll`.

Reply:
@MortalHappiness  Yes, you're right in that there are some `It` blocks that don't create folders, but I think it's fine to call `os.RemoveAll` to ensure that all `It` block start with the same state, or is `os.RemoveAll` too much?

Reply:
What I mean is that if `expectedDirPath` is not declared within an `It` block, its value will be the one from the previous `It`. As a result, `AfterEach` will attempt to delete a non-existent folder. Although it won't throw an error, it feels somewhat inappropriate. Perhaps we could set `expectedDirPath` to an empty string in `It` blocks where it isn't used and add a condition in `AfterEach` to execute only when `expectedDirPath` is not empty? Alternatively, since not every `It` requires folder deletion, we could skip using `AfterEach` and instead defer the delete operation directly in the `It` blocks where folder deletion is needed.

Reply:
Oh okay, I understand what you mean. Apologies for misinterpreting what you said. I agree that it does seem redundant to delete non-existing folders. I will go with the approach which it defers the delete op in the `It` block only for where deletion is needed.

Reply:
@MortalHappiness PTAL!